### PR TITLE
feat: Add plan-based quotas for GitHub Vector Stores with transaction-safe enforcement

### PR DESCRIPTION
**Pull request by shige**
## Overview

This PR introduces plan-based quotas and access controls for GitHub Vector Stores to prevent overuse and align feature availability with subscription tiers. The changes enforce repository registration limits based on team plans (Free: 0, Pro: 3, Team: 10, Internal: 100) and provide clear user feedback when limits are reached or feature access is unavailable.

## Related Issue
Part of https://github.com/giselles-ai/giselle/issues/2061

### Backend
- **New quota service module** (`gitHubVectorStoreQuota.ts`):
  - Provides helpers to check plan availability and enforce limits
  - Includes comprehensive unit tests for all plan scenarios
- **Registration enforcement** (`registerRepositoryIndex`):
  - Implements serialized transaction with row locking to prevent race conditions
  - Validates plan availability and enforces per-team limits
  - Returns specific error messages for different failure scenarios

### Frontend
- **UI/UX improvements**:
  - Registration dialog can be disabled with explanatory tooltips
  - Repository list displays usage statistics (used/limit)
  - Shows alerts when at capacity
  - Displays locked state with upgrade CTA for plans without access
- **Component updates**:
  - Team plan data is fetched and passed through to vector store pages
  - RepositoryList component updated to handle new quota-related props

## Testing

- ✅ Unit tests added for quota service covering all plan types and edge cases
- ✅ Backend enforcement tested with concurrent request scenarios
- ✅ UI tested across different plan configurations to verify correct display states
- ✅ Error handling verified for limit exceeded and no access scenarios

## Review Notes

⚠️ **Breaking Changes:**
- Teams on the free plan can no longer register repositories (previously unlimited)
- The `RepositoryList` component now requires additional props (`hasAccess`, `maxStores`, `teamPlan`)
- Existing workflows that assumed unlimited registration may need adjustment

**Areas for review:**
- Transaction locking strategy in `registerRepositoryIndex` - please verify the approach for preventing race conditions
- UX copy for error messages and upgrade CTAs
- Component prop changes and their potential impact on other parts of the codebase

## Related Issues

_No specific issue referenced - please link if applicable_

### CODE REVIEW

File: apps/studio.giselles.ai/app/(main)/settings/team/vector-stores/actions.ts [hunk 1]
```diff
@@ -145,26 +146,51 @@ export async function registerRepositoryIndex(
...
+				.limit(quota.maxStores)
+				.for("update");
+
+			if (existingRepositories.length >= quota.maxStores) {
+				return {
+					success: false,
+					error:
+						"You have reached the number of GitHub Vector Stores included in your plan.",
+				};
+			}
```
[the remark this answers is not in the document]
Comment (shige) [reply apps/studio.giselles.ai/app/(main)/settings/team/vector-stores/actions.ts:201]: fixed adda98e2379f5da31b4198c3d5f3bf1c2ae5adbf

**Review by toyamarinyon — APPROVED** [review]

**Comment by shige** [comment]
Thank you! 🚀
